tachyon/crypto/commitments/mixed_matrix_commitment_scheme.h

```diff
@@ -0,0 +1,45 @@
+#ifndef TACHYON_CRYPTO_COMMITMENTS_MIXED_MATRIX_COMMITMENT_SCHEME_H_
+#define TACHYON_CRYPTO_COMMITMENTS_MIXED_MATRIX_COMMITMENT_SCHEME_H_
+
+#include <utility>
+#include <vector>
+
+#include "third_party/eigen3/Eigen/Core"
+
+#include "tachyon/crypto/commitments/mixed_matrix_commitment_scheme_traits_forward.h"
+#include "tachyon/math/matrix/matrix_types.h"
+
+namespace tachyon::crypto {
+
+template <typename Derived>
+class MixedMatrixCommitmentScheme {
+ public:
+  using Field = typename MixedMatrixCommitmentSchemeTraits<Derived>::Field;
+  using Commitment =
+      typename MixedMatrixCommitmentSchemeTraits<Derived>::Commitment;
+
+  [[nodiscard]] bool Commit(const std::vector<Field>& vector,
+                            Commitment* result) {
+    math::RowMajorMatrix<Field> matrix(vector.size(), 1);
+    for (size_t i = 0; i < vector.size(); ++i) {
+      matrix(i, 0) = vector[i];
+    }
```

Comment:
Can't we commit in batch?

Reply:
Do you mean that does this scheme support Batch? Then the answer would be it always try to commit in batch since it accepts the vector of the matrices.

Reply:
@chokobole I was asking if we could do this for loop in parallel!

Reply:
Oh yes it's possible. I'll do that :)